test/e2e/e2e_test.go

```diff
@@ -52,8 +53,9 @@ func TestEndToEnd(t *testing.T) {
 	SetDefaultConsistentlyDuration(30 * time.Second)
 	SetDefaultConsistentlyPollingInterval(1 * time.Second)
 
-	if junitDir := os.Getenv("JUNIT_DIRECTORY"); junitDir != "" {
-		junitReporter := reporters.NewJUnitReporter(path.Join(junitDir, fmt.Sprintf("junit_e2e_%02d.xml", config.GinkgoConfig.ParallelNode)))
+	// always configure a junit report when ARTIFACTS_DIR has been set
+	if artifactsDir := os.Getenv("ARTIFACTS_DIR"); artifactsDir != "" {
```

Comment:
JUnit reports are now created implicitly whenever the $ARTIFACTS_DIRECTORY has been specified/non-empty.